src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/ClassWriter.java

```diff
@@ -97,7 +96,7 @@ public ClassWriter(HtmlConfiguration configuration, TypeElement typeElement,
         this.classTree = classTree;
 
         pHelper = new PropertyUtils.PropertyHelper(configuration, typeElement);
-        tocBuilder = new ListBuilder(HtmlTree.OL(HtmlStyle.tocList));
+        tableOfContents = new TableOfContents(this);
```

Comment:
Is there a reason to do this here (and also in `ModuleWriter` and `PackageWriter` etc) rather than in the (shared) super-constructor for `HtmlDocletWriter` ?

Reply:
The reason for doing it in the concrete writer classes is that only some `HtmlDocletWriter` subclasses have a table of contents, and I don't think there's an easy way to figure out in the super-constructor which instances need a table of contents and which don't.